packages/core/src/lib/ShortcutInfo.ts

```diff
@@ -33,15 +33,22 @@ export class ShortcutInfo {
    * @param {string} name
    * @param {string} shortName
    * @param {string} url target Url for when the shortcut is clicked
-   * @param {string} chosenIconUrl Url for the icon
+   * @param {string} chosenIconUrl Url for the icon with an "any" purpose
+   * @param {string} chosenMaskableIconUrl Url for the icon with a maskable purpose
+   * @param {string} chosenMonochromeIconUrl Url for the icon with a monochrome purpose
    */
   constructor(readonly name: string, readonly shortName: string, readonly url: string,
-        readonly chosenIconUrl: string) {
+    readonly chosenIconUrl?: string, readonly chosenMaskableIconUrl?: string,
```

Comment:
If I understood correctly, creating a `ShortcutInfo` where both `chosenIconUrl` and `monoChromeIconUrl` are `undefined` would mean we have an invalid instance? It's probably worth checking this in the constructor and preventing creating an invalid instance.